tests/addr.rs

```diff
@@ -0,0 +1,77 @@
+use std::net::{Ipv4Addr, Ipv6Addr, SocketAddr, SocketAddrV4, SocketAddrV6};
+
+use async_std::net::ToSocketAddrs;
+use async_std::task;
+
+fn tsa<A: ToSocketAddrs>(a: A) -> Result<Vec<SocketAddr>, String> {
+    let socket_addrs = task::block_on(a.to_socket_addrs());
+    match socket_addrs {
+        Ok(a) => Ok(a.collect()),
+        Err(e) => Err(e.to_string()),
+    }
+}
+
+pub fn sa4(a: Ipv4Addr, p: u16) -> SocketAddr {
```

Comment:
Still you did not test ToSocketAddrs::to_socket_addrs for (Ipv4Addr, u16). This is a useless function to create SocketAddr